src/member.ts

```diff
@@ -102,3 +102,38 @@ export async function deleteMember(
     didSucceed: true,
   };
 }
+
+export async function updateMember(
+  memberId: string,
+  memberData: PatchMemberSchema,
+  options: ServiceOptions,
+): Promise<Result<undefined, MemberProblemType>> {
+  const validationFailure = validateMemberData(memberData, options);
+  if (validationFailure !== undefined) {
+    return { didSucceed: false, reason: validationFailure };
+  }
+
+  const role = memberData.role && ROLE_MAPPING[memberData.role];
```

Comment:
```suggestion
  const role = memberData.role ?? ROLE_MAPPING[memberData.role];
```

That way, `role` type becomes `MemberRole | null`. (If I'm not mistaken, right now it's `MemberRole | false`)